crypto/src/main/java/org/hyperledger/besu/crypto/NodeKey.java

```diff
@@ -14,16 +14,38 @@
  */
 package org.hyperledger.besu.crypto;
 
-import org.hyperledger.besu.crypto.SECP256K1.PublicKey;
-import org.hyperledger.besu.crypto.SECP256K1.Signature;
-
 import org.apache.tuweni.bytes.Bytes32;
 
-public interface NodeKey {
+public class NodeKey {
+
+  private final SecurityModule crypto;
+
+  public NodeKey(final SecurityModule crypto) {
+    this.crypto = crypto;
+  }
+
+  public SECP256K1.Signature sign(final Bytes32 dataHash) {
+    org.hyperledger.besu.crypto.Signature signature = crypto.sign(dataHash);
```

Comment:
This can be now "non-quailfied" as `Signature signature = crypto.sign(dataHash);`

Reply:
done